--- metro-saaj-3.0.4/saaj-ri/src/main/java/com/sun/xml/messaging/saaj/soap/impl/ElementImpl.java	2021-06-29 10:22:21.541666957 +0200
+++ metro-saaj-3.0.4/saaj-ri/src/main/java/com/sun/xml/messaging/saaj/soap/impl/ElementImpl.java	2021-06-29 14:13:55.852768413 +0200
@@ -640,7 +657,11 @@
         Node child = getFirstChild();
         while (child != null) {
             if (child instanceof Element) {
-                return (Element) soapDocument.find(child);
+		try{
+                	return (Element) soapDocument.find(child);
+		}catch(IllegalArgumentException iax){
+			return (Element) child; //come in saaj 1.3.28
+		}
             }
             child = child.getNextSibling();
         }
