diff --git src/main/java/com/github/erosb/kappa/operation/validator/util/convert/XmlConverter.java src/main/java/com/github/erosb/kappa/operation/validator/util/convert/XmlConverter.java
index b74fa9d..8a8d833 100644
--- src/main/java/com/github/erosb/kappa/operation/validator/util/convert/XmlConverter.java
+++ src/main/java/com/github/erosb/kappa/operation/validator/util/convert/XmlConverter.java
@@ -117,6 +117,10 @@ class XmlConverter {
   }
 
   private JsonNode unwrap(OAIContext context, final Schema schema, final JsonNode content, final String defaultKey) {
+    if (content == null) {
+      return content;
+    }
+
     Xml xmlConf = schema.getXml();
 
     if (OAI3SchemaKeywords.TYPE_ARRAY.equals(schema.getSupposedType(context))) {
@@ -147,6 +151,10 @@ class XmlConverter {
   }
 
   private JsonNode getRenamedNode(final Xml xmlConf, final JsonNode content, final String defaultKey) {
+    if (content == null) {
+      return content;
+    }
+
     if (xmlConf != null && xmlConf.getName() != null) {
       return content.get(xmlConf.getName());
     } else if (defaultKey != null) {
