--- richfaces-ui-3.3.4.Final.orig/framework/impl/src/main/java/org/ajax4jsf/renderkit/RendererUtils.java	2022-11-10 19:29:53.399280468 +0100
+++ richfaces-ui-3.3.4.Final/framework/impl/src/main/java/org/ajax4jsf/renderkit/RendererUtils.java	2022-11-10 20:56:54.873263445 +0100
@@ -40,6 +40,7 @@
 import javax.faces.component.UIForm;
 import javax.faces.component.UIViewRoot;
 import javax.faces.component.ValueHolder;
+import javax.faces.context.ExternalContext;
 import javax.faces.context.FacesContext;
 import javax.faces.context.ResponseWriter;
 import javax.faces.convert.Converter;
@@ -210,6 +211,8 @@
 		public static final String src_ATTRIBUTE = "src";
 
 		public static final String media_ATTRIBUTE = "media";
+		
+		public static final String nonce_ATTRIBUTE = "nonce";
 
 		// public static final String onreset_ATTRIBUTE = "onreset";
 		// attributes sets.
@@ -248,6 +251,7 @@
 
 		public static final String[] PASS_THRU_STYLES = { "style", "class", };
 
+		public static final String STYLE_ELEM = "style";
 		public static final String SPAN_ELEM = "span";
 		public static final String DIV_ELEM = "div";
 		public static final String SCRIPT_ELEM = "script";
@@ -287,6 +291,10 @@
 		
 		public static final String DT_ELEMENT = "dt";
 		public static final String DL_ELEMENT = "dl";
+		
+		/* CSP */
+		public static final String REQUEST_ATTRIBUTE_CSP_RANDOM_NONCE = "_csp_random_nonce";
+		public static final String HEADER_NAME_CONTENT_SECURITY_POLICY = "Content-Security-Policy";
 	}
 
 	/**
@@ -1159,4 +1167,20 @@
 			return null;
 		}
 	}
+	
+	public static String getCspNonceValue(FacesContext context) {
+		ExternalContext extCtx = context.getExternalContext();
+		if(extCtx != null) {
+			String cspValue = (String) extCtx.getRequestMap().get(HTML.REQUEST_ATTRIBUTE_CSP_RANDOM_NONCE);
+			return cspValue;
+		}
+		
+		return null;
+	}
+	
+	public static String getCssId(String componentId) {
+		if(componentId == null) return null;
+		
+		return componentId.replace(":", "-").replace(".", "-");
+	}
 }
